### ISSUE TITLE
Adding metadata fails when downloading YouTube .weba audio file format.

### ISSUE THREAD

**Issue by DAM1173** [yt-dlp/yt-dlp#5973]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2023.01.02** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

When downloading music from YouTube in weba audio file format and adding any metadata by using --embed-metadata, I get the following error, for example:  yt_dlp.postprocessor.ffmpeg.FFmpegPostProcessorError: file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.temp.weba: Invalid argument.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
daw@Mint:~$ yt-dlp -f 251 -vU --rm-cache-dir --output "~/Muziek/Youtube-dl/%(title)s.%(ext)s" --embed-metadata https://www.youtube.com/watch?v=2Q_ZzBGPdqE
[debug] Command-line config: ['-f', '251', '-vU', '--rm-cache-dir', '--output', '~/Muziek/Youtube-dl/%(title)s.%(ext)s', '--embed-metadata', 'https://www.youtube.com/watch?v=2Q_ZzBGPdqE']
[debug] User config: []
[debug] System config: []
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.01.02 [d83b0ad] (linux_exe)
[debug] Python 3.10.8 (CPython x86_64 64bit) - Linux-5.15.0-56-generic-x86_64-with-glibc2.31 (OpenSSL 3.0.7 1 Nov 2022, glibc 2.31)
[debug] exe versions: ffmpeg N-109474-gc94988a781-20230103 (setts), ffprobe N-109474-gc94988a781-20230103, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1754 extractors
Removing cache dir /home/daw/.cache/yt-dlp ..
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.01.02, Current version: 2023.01.02
yt-dlp is up to date (2023.01.02)
[youtube] Extracting URL: https://www.youtube.com/watch?v=2Q_ZzBGPdqE
[youtube] 2Q_ZzBGPdqE: Downloading webpage
[youtube] 2Q_ZzBGPdqE: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] 2Q_ZzBGPdqE: Downloading 1 format(s): 251
[debug] Invoking http downloader on "https://rr5---sn-5hnekn7z.googlevideo.com/videoplayback?expire=1673016601&ei=ueC3Y5qUMZDWgQfH06jgAQ&ip=217.120.11.174&id=o-AI7eoitD3kFb_kwGMEu9iD-xrMt949BgosMix9Qp-2_b&itag=251&source=youtube&requiressl=yes&mh=P1&mm=31%2C26&mn=sn-5hnekn7z%2Csn-4g5lznls&ms=au%2Conr&mv=m&mvi=5&pl=22&gcr=nl&initcwndbps=2630000&spc=zIddbBPvoLcsz6BKrh_CfTu36trc1uI&vprv=1&svpuc=1&mime=audio%2Fwebm&gir=yes&clen=2184885&dur=139.181&lmt=1670072605882574&mt=1672994537&fvip=4&keepalive=yes&fexp=24007246&c=ANDROID&txp=4532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cgcr%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgBX8ALSEJ3EROTgVeoav4snfvBd_d-iqcpmASWr10zMsCIQCs-OY0Go5rtwCkB_uCbWhFCJ-xL8aoJK5jGdrtHq9egg%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgc4yNEfiYbB_iWNX4mipQeQbl4aXWV5cyXP1trE3sbUICIQD0Clo3-Eo-gOKQIs71zeb13PPHYo5NrQe3Kzrg2Xutpg%3D%3D"
[download] Destination: /home/daw/Muziek/Youtube-dl/The Beatles - Help!.weba
[download] 100% of    2.08MiB in 00:00:00 at 6.85MiB/s
[Metadata] Adding metadata to "/home/daw/Muziek/Youtube-dl/The Beatles - Help!.weba"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i 'file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.weba' -map 0 -dn -ignore_unknown -c copy -write_id3v1 1 -metadata 'title=The Beatles - Help!' -metadata date=20180621 -metadata 'description=Music video by The Beatles performing Help!. © 2015 Calderstone Productions Limited (a division of Universal Music Group) / Subafilms Ltd

http://vevo.ly/Bv0s2I' -metadata 'synopsis=Music video by The Beatles performing Help!. © 2015 Calderstone Productions Limited (a division of Universal Music Group) / Subafilms Ltd

http://vevo.ly/Bv0s2I' -metadata 'purl=https://www.youtube.com/watch?v=2Q_ZzBGPdqE' -metadata 'comment=https://www.youtube.com/watch?v=2Q_ZzBGPdqE' -metadata artist=TheBeatlesVEVO -movflags +faststart 'file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.temp.weba'
[debug] ffmpeg version N-109474-gc94988a781-20230103 Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 12.2.0 (crosstool-NG 1.25.0.90_cf9beb1)
  configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-ffbuild-linux-gnu- --arch=x86_64 --target-os=linux --enable-gpl --enable-version3 --disable-debug --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-lzma --enable-fontconfig --enable-libvorbis --enable-opencl --enable-libpulse --enable-libvmaf --enable-libxcb --enable-xlib --enable-amf --enable-libaom --enable-libaribb24 --enable-avisynth --enable-chromaprint --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libkvazaar --enable-libass --enable-libbluray --enable-libjxl --enable-libmp3lame --enable-libopus --enable-mbedtls --enable-librist --enable-libssh --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --disable-libmfx --enable-libvpl --enable-openal --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --disable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --enable-libdrm --enable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags=-pie --extra-libs='-ldl -lgomp' --extra-version=20230103
  libavutil      57. 43.100 / 57. 43.100
  libavcodec     59. 56.100 / 59. 56.100
  libavformat    59. 34.102 / 59. 34.102
  libavdevice    59.  8.101 / 59.  8.101
  libavfilter     8. 53.100 /  8. 53.100
  libswscale      6.  8.112 /  6.  8.112
  libswresample   4.  9.100 /  4.  9.100
  libpostproc    56.  7.100 / 56.  7.100
Input #0, matroska,webm, from 'file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.weba':
  Metadata:
    encoder         : google/video-file
  Duration: 00:02:19.18, start: -0.007000, bitrate: 125 kb/s
  Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
[NULL @ 0x55d7ad343c80] Unable to find a suitable output format for 'file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.temp.weba'
file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.temp.weba: Invalid argument

ERROR: Postprocessing: file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.temp.weba: Invalid argument
Traceback (most recent call last):
  File "yt_dlp/YoutubeDL.py", line 3307, in process_info
  File "yt_dlp/YoutubeDL.py", line 3486, in post_process
  File "yt_dlp/YoutubeDL.py", line 3468, in run_all_pps
  File "yt_dlp/YoutubeDL.py", line 3446, in run_pp
  File "yt_dlp/postprocessor/common.py", line 24, in run
  File "yt_dlp/postprocessor/common.py", line 129, in wrapper
  File "yt_dlp/postprocessor/ffmpeg.py", line 705, in run
  File "yt_dlp/postprocessor/ffmpeg.py", line 323, in run_ffmpeg_multiple_files
  File "yt_dlp/postprocessor/ffmpeg.py", line 361, in real_run_ffmpeg
yt_dlp.postprocessor.ffmpeg.FFmpegPostProcessorError: file:/home/daw/Muziek/Youtube-dl/The Beatles - Help!.temp.weba: Invalid argument
```

**Comment by bashonly** [comment]
This is a bug introduced in version 2023.01.02 which causes webm audio to be incorrectly extracted with a `.weba` extension (and `.weba` is not supported by ffmpeg). It has been fixed in the master branch (d80ca5deaa46db6e498399bb04a72a4c10ee8e22), and the fix will be included in the next release.

A workaround for now is to use `-S "ext"` to prefer mp4/m4a formats over webm/weba, but this is workaround probably not preferable for you if you're specifically selecting a webm format.

If you don't want to wait for the next release, you can [install the master branch with PIP](https://github.com/yt-dlp/yt-dlp/wiki/Installation#with-pip), or try the [unofficial daily builds](https://github.com/ytdl-patched/yt-dlp)

**Comment by DAM1173** [comment]
I use the latest custom-built ffmpeg for yt-dlp at https://github.com/yt-dlp/FFmpeg-Builds/releases/download/latest/ffmpeg-master-latest-linux64-gpl.tar.xz
